example/glue/ConfigurationDelegate.hpp

```diff
@@ -0,0 +1,146 @@
+
+/**
+ * Client language delegate for GC configuration must be implemented by all GC clients.
+ */
+class MM_ConfigurationDelegate
+{
+/*
+ * Member data and types
+ */
+private:
+	const MM_GCPolicy _gcPolicy;
+
+protected:
+public:
+
+/*
+ * Member functions
+ */
+private:
+protected:
+
+public:
+	/**
+	 * Initialize the delegate.
+	 * @param env The calling thread environment
+	 * @param writeBarrierType The write barrier type selected for the GC configuration
+	 * @param allocationType The allocation type selected for the GC configuration
+	 * @return false if initialization fails
+	 */
+	bool
+	initialize(MM_EnvironmentBase* env, MM_GCWriteBarrierType writeBarrierType, MM_GCAllocationType allocationType)
+	{
+		return true;
+	}
+
+	/**
+	 * Tear down the delegate, releasing any resources held.
+	 * @param env The calling thread environment
+	 */
+	void
+	tearDown(MM_EnvironmentBase* env)
+	{
+		return;
+	}
+
+	/**
+	 * This method is called just after the heap has been initialized. Delegate may perform
+	 * any additional heap-related initialization at this point.
+	 * @param env The calling thread environment
+	 * @return false if heap initialization fails
+	 */
+	bool
+	initializeHeap(MM_EnvironmentBase* env)
+	{
+		return true;
+	}
+
+	/**
+	 * The OMR GC preallocates a pool of MM_EnvironmentBase subclass instances on startup. This method is called
+	 * to allow GC clients to determine the number of pooled instances to preallocate.
+	 * @param env The calling thread environment
+	 * @return The number of pooled instances to preallocate, or 0 to use default number
+	 */
+	uint32_t
+	getInitialNumberOfPooledEnvironments(MM_EnvironmentBase* env)
+	{
+		return 0;
+	}
+
+	/**
+	 * The OMR segregated GC requires GC clients to specify a distribution of segment sizes with which
+	 * it will partition the heap. This method is called to retrieve the size distribution metadata.
+	 * @param env The calling thread environment
+	 * @return A pointer to segregated heap size distribution metadata, or NULL if not using segregated heap
+	 */
+	OMR_SizeClasses *getSegregatedSizeClasses(MM_EnvironmentBase *env)
+	{
+		return NULL;
+	}
+
+	/**
+	 * Perform GC client-specific initialization for a new MM_EnvironmentBase subclass instance.
+	 * @param env The thread environment to initialize
+	 * @return false if environment initialization fails
+	 */
+	bool
+	initializeEnvironment(MM_EnvironmentBase* env)
+	{
+		env->_envLanguageInterface = MM_EnvironmentLanguageInterfaceImpl::newInstance(env);
+		return (NULL != env->_envLanguageInterface);
+	}
+
+	/**
+	 * The OMR GC preallocates a pool of collector threads. This method is called to allow GC clients
+	 * to determine the maximum number of concurrent collector threads to run during GC cycles.
+	 * @param env The calling thread environment
+	 * @return The maximum number of GC threads to allocate for the runtime GC
+	 */
+	uintptr_t getMaxGCThreadCount(MM_EnvironmentBase* env)
+	{
+		return 1;
```

Comment:
That is actually the Slave threads and not the concurrent threads.  Can you update the documentation to reflect this?  Instead of using slave threads you can talk about this number being used to limit the number of threads created by the paralleldispatcher or something like that.